File: pset2/vigenere.c
/****************************************************************************
 * vigenere.c
 * Programmeren 01 - pset 02 - exercise 03
 * By Maxim Stomphorst
 * 
 * This converts text to ciphertext with a key that the user give's.
 * But the key is a word in this case.
***************************************************************************/

#include <cs50.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <ctype.h>

// the key is given in Command Line Argument (CLA)
int main(int argc, string argv[])
{
    // check if there are 2 CLA
    if (argc != 2)
    {
        // telling the user how the CLA works
        printf("Usage: ./vigenere key\n");
        return 1;
    }
    
    // gowing through 2nd CLA posission for posission
    for (int i = 0 ; i < strlen(argv[1]); i++)
    {
        // checking if ith posission is alphabetical
        if (isalpha(argv[1][i]))
        {
            // converting ith posission to lower case letter
            argv[1][i] = tolower(argv[1][i]);
        }
        else
        {
            // telling the user what to do
            printf("the key must be alphabetical");
            return 1;
        }
    }
    
    // asking user for input
    printf("plaintext: ");
    string text = get_string();

    // converting CLA input to string
    string key = (argv[1]);
    // counting string lenghts
    int text_l = strlen(text);
    int key_l = strlen(key);
    printf("ciphertext: ");
    
    // gowing through text char for char (i)
    // gowing throug key char for char (j) only j++ when used
    for (int i = 0, j = 0; i < text_l; i++)
    {
        // checking if ith char is upper
        if (isupper(text[i]))
        {
            // convert ascii(upper) to alphabet count - 64
            // wrap around key j % key_l  
            // convert ascii(lower) to alphabet count - 97
            // wrap around alphabet % 26 
            // convert alphabet count back to ascii(upper)
            printf("%c",(((text[i] - 'A') + (key[j % key_l] - 'a') ) % 26 ) + 'A');
            // j used so j++ (independent variable for key posission) 
            j++;
        }
        // checking if ith char is lower
        else if (islower(text[i]))
        {
            // every char is lower - 97 no convertion too / from upper
            printf("%c",(((text[i] - 'a') + (key[j % key_l] - 'a') ) % 26 ) + 'a');
            j++;
        }
        else
        {
            // downt convert ith char directly print 
            printf("%c", text[i]);
        }
    }
    printf("\n");
    return 0;
}

File: pset6/port/vigenere.py
# vigenere.py
# Programmeren 02 - pset 06 - exercise 03
# By Maxim Stomphorst
#
# This converts text to ciphertext with a key that the user give's.
# But the key is a word in this case.

import cs50
import sys

# checking if the user gave a key
if not len(sys.argv) == 2:
    print("Usage: ./vigenere key")
    exit(1)

# check if the key alphabetical
if not sys.argv[1].isalpha():
    print("the key must be alphabetica!")
    exit(2)

# converting key too lowercase
key = sys.argv[1].lower()

# asking user for input
print("Plaintext: ", end="")
plaintext = cs50.get_string()

# indipendent counter for iterating over the key
j = 0
key_lenght = len(key)
print("ciphertext: ", end="")

# gowing through plaintext char for char (i)
# gowing throug key char for char (j) only j++ when used
for letter in plaintext:
    # checking if ith char is upper
    if letter.isupper():
        # convert ascii(upper) to alphabet count (0...26) | - 64
        # wrap around key | j % key_lenght
        # convert ascii(lower) to alphabet count (0...26) | - 97
        # wrap around alphabet | % 26
        # convert alphabet count back to ascii(upper) | + 64
        print(chr(((ord(letter) - ord('A') + (ord(key[j % key_lenght]) -
                                              ord('a'))) % 26) + ord('A')), end="")
        j += 1

    # checking if ith char is lower
    elif letter.islower():
        # every char is lower - 97 no convertion too / from upper
        print(chr(((ord(letter) - ord('a') + (ord(key[j % key_lenght]) -
                                              ord('a'))) % 26) + ord('a')), end="")
        j += 1

else:
    # downt convert
    print(letter, end="")

print()
exit(0)

